### PR TITLE
[OY2-9913] Add measure AMR-AD + Add "Totals" to Rate component

### PR DESCRIPTION
**Pull request by jphechter**
## Purpose

Add the AMR-AD measure.

#### Linked Issues to Close

[OY2-9913](https://qmacbis.atlassian.net/jira/software/c/projects/OY2/boards/208?modal=detail&selectedIssue=OY2-9913)

## Approach

- Add the AMR-AD measure
  - create a new measure
  - created net-new functionality to support the desired experience
- Updated QMR.Rate component
  - added support for a new optional "Totals" N/D/R section
  - added the field label for each N/D/R to better label data stored in DB
- Updates to AMR-AD's OMS section
  - Calculating totals required a re-work of Rate component implementation
- Styling changes to the validation error messaging
  - Moved inside the MeasureWrapper to match other content width

## Learning

We learned... a lot.

## Assorted Notes/Considerations

We added notes to our implementation of QMR.Rate to show how Total works, but maybe this could use other notes if its going to be implemented for another measure.

#### Pull Request Creator Checklist

- [X] This PR has an associated issue or issues.
- [X] The associated issue(s) are linked above.
- [X] This PR meets all acceptance criteria for those issues.
- [x] This PR and linked issue(s) are adequately documented
- [x] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [x] Someone has been assigned this PR.
- [X] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [x] This PR has an associated issue or issues.
- [x] The associated issue(s) are linked above.
- [x] This PR meets all acceptance criteria for those issues.
- [x] This PR and linked issue(s) are adequately documented
- [x] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

File: services/ui-src/src/components/MeasureWrapper/index.tsx [hunk 1]
```diff
@@ -279,21 +279,21 @@ export const MeasureWrapper = ({
...
+                    alertStatus="error"
+                    alertTitle={`${error.errorLocation} Error`}
+                    alertDescription={error.errorMessage}
+                    close={() => {
+                      const newErrors = [...errors];
+                      newErrors.splice(index, 1);
+                      setErrors(newErrors);
+                    }}
+                  />
+                ))}
```
Comment (jphechter) [inline services/ui-src/src/components/MeasureWrapper/index.tsx:295]: This is just a styling change. By moving this inside the above section, validation message max-width matches the rest of the page content.

Comment (charmcitygavin) [reply services/ui-src/src/components/MeasureWrapper/index.tsx:295]: Before the change:
Screen Shot 2022-02-17 at 3 33 52 PM

Comment (jphechter) [reply services/ui-src/src/components/MeasureWrapper/index.tsx:295]: After the change:
Screen Shot 2022-02-17 at 3 31 46 PM

**Review by jdinh8124 — APPROVED** [review]
Great job, I have a small comment and make sure to check failing steps like the deploy/clearing code climate. Let me know if you need help.

**Review by jdinh8124 — DISMISSED** [review]

File: services/ui-src/src/measures/2021/AMRAD/questions/PerformanceMeasure.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      label: "Total (Ages 19 to 64)",
+      denominator: "",
+      numerator: "",
+      rate: "",
+      id: 3,
+      isTotal: true,
+    },
+  ];
+
+  // TODO: The Performance Measures should total automagically
```
Comment (jdinh8124) [inline services/ui-src/src/measures/2021/AMRAD/questions/PerformanceMeasure.tsx:46]: Will this be a todo in a different measure or is this addressed already?

File: services/ui-src/src/measures/2021/AMRAD/questions/MeasurementSpecification.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
+// import * as CUI from "@chakra-ui/react";
+import * as QMR from "components";
```
Comment (jdinh8124) [inline services/ui-src/src/measures/2021/AMRAD/questions/MeasurementSpecification.tsx:2]: Can we use the CommonQuestions version of this file?

File: services/ui-src/src/measures/2021/AMRAD/index.tsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+            showPersistentAsthma51To64 ||
+            showPersistentAsthmaTotal ||
+            showOtherPerformanceMeasureRates) && (
+            <Q.OptionalMeasureStratification
+              ageGroups={ageGroups}
+              deviationConditions={{
+                showPersistentAsthma19To50,
+                showPersistentAsthma51To64,
+                showPersistentAsthmaTotal,
+                // showOtherPerformanceMeasureRates,
```
Comment (jdinh8124) [inline services/ui-src/src/measures/2021/AMRAD/index.tsx:117]: cleanup if possible

File: services/ui-src/src/measures/2021/AMRAD/questions/PerformanceMeasure.tsx (hunk 2 above)
Comment (jphechter) [reply services/ui-src/src/measures/2021/AMRAD/questions/PerformanceMeasure.tsx:46]: Old TODO. This is referencing the changes made to the Rate component.

Comment (charmcitygavin) [reply services/ui-src/src/measures/2021/AMRAD/questions/PerformanceMeasure.tsx]: Removed.

File: services/ui-src/src/measures/2021/AMRAD/questions/MeasurementSpecification.tsx (hunk 3 above)
Comment (charmcitygavin) [reply services/ui-src/src/measures/2021/AMRAD/questions/MeasurementSpecification.tsx]: Great idea. Updated! 👍🏻

File: services/ui-src/src/measures/2021/AMRAD/index.tsx (hunk 4 above)
Comment (charmcitygavin) [reply services/ui-src/src/measures/2021/AMRAD/index.tsx]: Cleaned up 👍🏻

**Review by jdinh8124 — DISMISSED** [review]

**Review by daniel-belcher — APPROVED** [review]
Added oms validation and everything elselooks good.

**Comment by jphechter** [comment]
OMS Validation looks good to me!

**Review by abelzer212 — APPROVED** [review]

**Review by abelzer212 — APPROVED** [review]

**Review by abelzer212 — APPROVED** [review]

**Pull request merged**